packages/core/src/Page/PageVueServerRenderer.js

```diff
@@ -0,0 +1,142 @@
+
+/**
+ * Compiles page into Vue Application to get the page render function and places
+ * it into a script so that the browser can retrieve the page render function to
+ * render the page during Vue mounting.
+ *
+ * This is to avoid the overhead of compiling the page into Vue application
+ * on the client's browser (alleviates FOUC). It is also the pre-requisite to enable SSR.
+ *
+ * @param content Page content to be compiled into Vue app
+ */
+async function compileVuePageAndCreateScript(content, pageConfig, pageAsset) {
+  // Compile Vue Page
+  const compiledVuePage = VueCompiler.compileToFunctions(content);
+
+  // Set up script content
+  const outputContent = `
+    var pageVueRenderFn = ${compiledVuePage.render};
+    var pageVueStaticRenderFns = [${compiledVuePage.staticRenderFns}];
+  `;
+
+  // Get script file name
+  const pageHtmlFileName = path.basename(pageConfig.resultPath, '.html');
+  const scriptFileName = `${pageHtmlFileName}.page-vue-render.js`;
+
+  /*
+   * Add the script file path for this page's render function to the page's assets (to populate page.njk).
+   * The script file path is the same as the page's file path.
+   */
+  pageAsset.pageVueRenderJs = scriptFileName;
+
+  // Get script's absolute file path to output script file
+  const dirName = path.dirname(pageConfig.resultPath);
+  const filePath = path.join(dirName, scriptFileName);
+
+  await fs.outputFile(filePath, outputContent);
+
+  return compiledVuePage;
+}
+
+/**
+ * Renders Vue page app into html string (Vue SSR).
+ * This function will install the MarkBindVue plugin and render the built Vue page content into html string.
+ */
+async function renderVuePage(compiledVuePage) {
+  const { MarkBindVue, appFactory } = bundle;
+
+  /*
+   * Each installation of plugin pollutes the global scope of Vue.
+   * Thus, we use a fresh copy of Vue each time we install MarkBindVue to prevent the global pollution.
+   * This prevents the old plugin from affecting the new plugin installation (in development mode).
+   */
+  const LocalVue = createLocalVue();
+  LocalVue.use(MarkBindVue);
+
+  const VueAppPage = new LocalVue({
+    render(createElement) {
+      return compiledVuePage.render.call(this, createElement);
+    },
+    staticRenderFns: compiledVuePage.staticRenderFns,
+    ...appFactory(),
+  });
+
+  let renderedContent = await renderToString(VueAppPage);
+
+  /*
+   * https://github.com/vuejs/vue/blob/dev/packages/vue-server-renderer/basic.js
+   *
+   * vue-server-renderer automatically escapes various characters during rendering for security reasons.
+   * We un-escape these characters as there shouldn't be any security concerns in our use-case.
+   *
+   * We only un-escape & and " for now so that the script redirect for userGuide/devGuide will work.
+   * Un-escaping < and/or > will lead to hydration issues so we do not do it for now.
+   *
+   * To be discussed with Zeyu for better workarounds(?)
```

Comment:
let's remove this comment line as well.

But this is ok for this mvp -- the long term solution should probably be to hoist all script / style tags downward, which also avoids possible hydration issues if any of these script tags manipulates the dom before vue setup.

Reply:
Sure thing, can I double check if you are just referring to line 94? or the whole comment chunk?

Reply:
Just line 94 -- but either would be ok as well since this has to be fixed before release.